From e1a6912fac339601da4f02f67645229b9ed32041 Mon Sep 17 00:00:00 2001
From: Daniel Cederman <cederman@gaisler.com>
Date: Fri, 4 Sep 2020 09:53:23 +0200
Subject: [PATCH 20/50] vxworks: Add VxWorks target

Adds define to signal that vendor is vxworks. This allows us to do some
custom changes when compiling the bare metal compiler for VxWorks.
 Mainly we do not want to create the builtin define "sparc".
---
 gcc/config.gcc                     | 4 ++++
 gcc/config/sparc/gaisler-vxworks.h | 1 +
 2 files changed, 5 insertions(+)
 create mode 100644 gcc/config/sparc/gaisler-vxworks.h

diff --git a/gcc/config.gcc b/gcc/config.gcc
index e250307083d..ddec8d3319e 100644
--- a/gcc/config.gcc
+++ b/gcc/config.gcc
@@ -3278,6 +3278,10 @@ sparc-*-elf*)
 		tm_file="${tm_file} sparc/bcc.h"
 		extra_options="${extra_options} sparc/bcc.opt"
 		;;
+	    sparc-vxwork*)
+		tmake_file="sparc/t-sparc"
+		tm_file="${tm_file} sparc/gaisler-vxworks.h"
+		;;
 	    *-leon-*)
 		tmake_file="sparc/t-sparc sparc/t-leon"
 		;;
diff --git a/gcc/config/sparc/gaisler-vxworks.h b/gcc/config/sparc/gaisler-vxworks.h
new file mode 100644
index 00000000000..0a512e7cac1
--- /dev/null
+++ b/gcc/config/sparc/gaisler-vxworks.h
@@ -0,0 +1 @@
+#define TARGET_GAISLER_VXWORKS 1
-- 
2.34.1

